lochness/db/models/remote_files.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+"""
+RemoteFile Model
+"""
+
+from datetime import datetime
+from typing import Dict
+
+from lochness import db
+
+
+class RemoteFile:
+    """
+    Represents a file on some remote file system.
+
+    Attributes:
+        file_path (Path): The path to the file.
+    """
+
+    def __init__(
+        self,
+        file_path: str,
+        remote_name: str,
+        hash_val: str,
+        last_checked: datetime,
+        remote_metadata: Dict[str, str],
+    ):
+        """
+        Initialize a RemoteFile object.
+
+        Args:
+            file_path (Path): The path to the file.
+            remote_name (str): The name of the remote system.
+            hash_val (str): The hash value of the file,
+                as provided by the remote system.
+            last_checked (datetime): The last time the file was checked.
+            remote_metadata (Dict[str, str]): Metadata about the file,
+        """
+        self.file_path = file_path
+        self.remote_name = remote_name
+        self.hash_val = hash_val
+        self.last_checked = last_checked
+        self.remote_metadata = remote_metadata
```

Comment:
We also want to add `self.remote_file_modifier` or `self.owner` (null and blank True) attribute to record who made the files on the source whenever the information is available

Reply:
Yep, those metadata will be on `self.metadata` as a dict. I chose a dict, as it allows more flexibility in modifying fields, without needing a schema change later.

So in our case, a sample remote_metadata dict can be:

```json
{
    "last_modified": "2024-07-29",
    "owner": "testUser"
}
```